pkg/commands/edit/add/addmetadata.go

```diff
@@ -135,13 +135,28 @@ func (o *addMetadataOptions) convertToMap(arg string) (map[string]string, error)
 			return nil, o.makeError(input, "empty key")
 		}
 		if len(kv) > 2 {
-			return nil, o.makeError(input, "too many colons")
-		}
-		if len(kv) > 1 {
+			// more than one colon found
+			// check if value is quoted
+			qc := strings.Index(input, ":\"")
+			if qc >= 1 && input[len(input)-1:] == "\"" {
+				//value is quoted
+				result[kv[0]] = input[qc+1:]
+			} else {
+				// value is not quoted, return error
+				return nil, o.makeError(input, "too many colons, quote the values")
+			}
+		} else if len(kv) == 2 {
 			result[kv[0]] = kv[1]
 		} else {
 			result[kv[0]] = ""
 		}
+
+		// remove quotes if value is quoted
+		if len(result[kv[0]]) > 0 &&
+			result[kv[0]][:1] == "\"" &&
+			result[kv[0]][len(result[kv[0]])-1:] == "\"" {
+			result[kv[0]] = result[kv[0]][1 : len(result[kv[0]])-1]
+		}
```

Comment:
Thanks so much for writing tests!

Suggest that you simplify this code by taking the form:
```
c := strings.Index(input, ":")
if c < 0 {
  return nil, fmt.Error("need k:v pair where v may be quoted, but got '%s'", input)
}
k := input[:c]
v := trimQuotes(input[c+1:])
...
```
where trimQuotes is as done at https://stackoverflow.com/a/44222891

Reply:
Done...

Reply:
Simplified the logic & moved the trim quotes logic to separate function.